libcaja-private/caja-file.c

```diff
@@ -3286,6 +3474,9 @@ caja_file_compare_for_sort (CajaFile *file_1,
 				result = compare_by_full_path (file_1, file_2);
 			}
 			break;
+		case CAJA_FILE_SORT_BY_EXTENSION:
+			result = compare_by_basename_segments (file_1, file_2);
+			break;
```

Comment:
All other sort criteria have a fallback just in case:
```c
if (result == 0) {
    result = compare_by_full_path (file_1, file_2);
}
```
It might make sense to add that to this as well to keep it consistent (unless there's a valid reason to not fall back to full path?)